src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -158,11 +168,11 @@ public void Dispose()
             _parametersCache.Dispose();
         }
 
-        private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        private async Task<bool> IsEnabledAsync<TContext>(FeatureDefinition featureDefinition, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
         {
```

Comment:
Usually for private methods where we are guaranteeing non-null elsewhere it is helpful to have a 

`Debug.Assert(featureDefinition != null);`

To set reader expectations that it is accounted for.